utils/forest_shim/src/address.rs

```diff
@@ -69,3 +69,16 @@ impl From<Address> for Address_v2 {
             .expect("Couldn't convert between FVM2 and FVM3 addresses")
     }
 }
+
+impl From<&Address> for Address_v2 {
+    fn from(other: &Address) -> Self {
+        Address_v2::from_bytes(&other.to_bytes())
+            .expect("Couldn't convert between FVM2 and FVM3 addresses")
```

Comment:
Maybe add some comments on why favor panic here than implementing `TryFrom` trait

Reply:
The two address versions are identical but the internals are hidden. If the conversion fails then Forest cannot continue in any way or form.

Reply:
@lemmih that makes sense. My suggestion is adding this to the code comment so that ppl won't wonder why not use `TryFrom` a few months later

Reply:
Added the comment. Hopefully, it's helpful enough.